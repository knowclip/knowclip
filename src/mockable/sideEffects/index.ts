import sideEffects from './module'
import { setUpMocks } from 'setUpMocks'

const mocked = setUpMocks('side-effect', sideEffects)

const { uuid, nowUtcTimestamp } = mocked

export { uuid, nowUtcTimestamp }
