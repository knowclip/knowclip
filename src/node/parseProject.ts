import { parse } from 'yaml'
import { readFile } from 'fs/promises'
import { ProjectJson } from '../types/Project'
import validateProject from './validateProject'

export type NormalizedProjectFileData = {
  project: ProjectFile
  media: MediaFile[]
  clips: Clip[]
  cards: Flashcard[]
  subtitles: SubtitlesFile[]
}

export const parseProjectJson = async <F extends FlashcardFields>(
  filePath: string
): AsyncResult<ProjectJson<F>> => {
  try {
    const docsText = (await readFile(filePath, 'utf8'))
      .split(/(?:^|\r?\n)(?:---|\.\.\.)\r?\n/)
      .filter((x) => x.trim())
    const errors: string[] = []
    const docs = docsText.map((text) => {
      try {
        const doc = parse(text, {
          maxAliasCount: -1,
        })
        return doc
      } catch (err) {
        console.error(err)
        errors.push(String(err))
        return null
      }
    })

    if (errors.length) return { errors }

    const [project, ...media] = docs

    const validation = validateProject(project, media)

    if (validation.errors)
      return {
        errors: Object.entries(validation.errors).map(
          ([sectionName, bigErrorMessage]) => {
            return `Invalid data for ${sectionName}:\n\n${bigErrorMessage}`
          }
        ),
      }

    return {
      value: {
        project,
        media,
      },
    }
  } catch (err) {
    return { errors: [String(err)] }
  }
}
