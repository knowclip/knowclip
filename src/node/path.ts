export { basename, extname, dirname, resolve, join } from 'path'
