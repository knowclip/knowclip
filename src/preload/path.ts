import { getPreloadModule } from './getPreloadModule'

const { basename, extname, dirname, resolve, join } =
  getPreloadModule('path') || require('path')

export { basename, extname, dirname, resolve, join }
