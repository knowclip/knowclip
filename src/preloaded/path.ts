import { getPreloadModule } from './getPreloadModule'

const { basename, extname, dirname, resolve, join } = getPreloadModule('path')

export { basename, extname, dirname, resolve, join }
