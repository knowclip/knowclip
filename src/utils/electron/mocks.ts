import helpers from './helpers'
import mockFunctions from '../../test/getFunctionMockers'

// IF THIS IS REERENCED OUTSIDE OF TESTS, this should use preload stuff, maybe on window?
// but i dont think it is
export const mockElectronHelpers =
  mockFunctions<typeof helpers>('electron-helper')
