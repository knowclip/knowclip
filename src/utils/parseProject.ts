import { parse } from 'yaml'
import { readFile } from 'preloaded/fs'
import { blankSimpleFields, blankTransliterationFields } from './newFlashcard'
import { parseFormattedDuration } from './formatTime'
import {
  ProjectJson,
  MediaJson,
  SubtitlesJson,
  EmbeddedSubtitlesJson,
} from '../types/Project'
import validateProject from './validateProject'
import { unescapeClozeFields } from './clozeField'

type NormalizedProjectFileData = {
  project: ProjectFile
  media: MediaFile[]
  clips: Clip[]
  cards: Flashcard[]
  subtitles: SubtitlesFile[]
}

export const parseProjectJson = async <F extends FlashcardFields>(
  filePath: string
): AsyncResult<ProjectJson<F>> => {
  try {
    console.log('reading', filePath)
    const docsText = (await readFile(filePath))
      .split(/(?:^|\n)(?:---|\.\.\.)\n/)
      .filter((x) => x.trim())
    const errors: string[] = []
    console.log('parsing', docsText)
    const docs = docsText.map((text) => {
      try {
        const doc = parse(text, {
          maxAliasCount: -1,
        })
        return doc
      } catch (err) {
        console.error(err)
        console.log('error:::', { text })
        errors.push(String(err))
        return null
      }
    })

    if (errors.length) return { errors }

    const [project, ...media] = docs

    const validation = validateProject(project, media)

    if (validation.errors)
      return {
        errors: Object.entries(validation.errors).map(
          ([sectionName, bigErrorMessage]) => {
            return `Invalid data for ${sectionName}:\n\n${bigErrorMessage}`
          }
        ),
      }

    return {
      value: {
        project,
        media,
      },
    }
  } catch (err) {
    return { errors: [String(err)] }
  }
}

export const normalizeProjectJson = <F extends FlashcardFields>(
  state: AppState,
  { project: projectJson, media: mediaJson }: ProjectJson<F>
): NormalizedProjectFileData => {
  const project: ProjectFile = {
    id: projectJson.id,
    createdAt: projectJson.createdAt,
    lastSaved: projectJson.timestamp,
    noteType: projectJson.noteType,
    name: projectJson.name,
    mediaFileIds: mediaJson.map((m) => m.id),
    type: 'ProjectFile',
    error: null,
  }
  const media: [MediaFile, () => Clip[], () => Flashcard[], SubtitlesFile[]][] =
    mediaJson.map((m) => {
      const subtitles = m.subtitles
        ? m.subtitles.map((s) => toMediaSubtitlesRelation(s))
        : []
      const subtitlesFiles: SubtitlesFile[] = (m.subtitles || []).map(
        (s): SubtitlesFile =>
          s.type === 'Embedded'
            ? {
                type: 'VttConvertedSubtitlesFile',
                id: s.id,
                streamIndex: s.streamIndex,
                parentId: m.id,
                parentType: 'MediaFile',
                chunksMetadata: s.chunksMetadata,
              }
            : {
                type: 'ExternalSubtitlesFile',
                id: s.id,
                name: s.name,
                parentId: m.id,
                chunksMetadata: s.chunksMetadata,
              }
      )
      const base: AudioFile = {
        name: m.name,
        type: 'MediaFile',
        id: m.id,
        parentId: projectJson.id,
        durationSeconds: parseFormattedDuration(m.duration).asSeconds(),
        format: m.format,
        subtitles,
        flashcardFieldsToSubtitlesTracks:
          m.flashcardFieldsToSubtitlesTracks || {},
        subtitlesTracksStreamIndexes: (m.subtitles || [])
          .filter((s): s is EmbeddedSubtitlesJson => s.type === 'Embedded')
          .map((s) => s.streamIndex),

        isVideo: false,
      }

      if ('width' in m) {
        return [
          {
            ...base,
            isVideo: true,
            width: m.width,
            height: m.height,
          },
          getMediaClipsGetter<F>(state, project, m),
          getMediaCardsGetter<F>(state, project, m),
          subtitlesFiles,
        ]
      }

      return [
        base,
        getMediaClipsGetter<F>(state, project, m),
        getMediaCardsGetter<F>(state, project, m),
        subtitlesFiles,
      ]
    })

  let clips: Clip[]
  let cards: Flashcard[]

  return {
    project,
    media: media.flatMap(([m]) => m),
    get clips() {
      clips = clips || media.flatMap(([, c]) => c())
      return clips
    },
    get cards() {
      cards = cards || media.flatMap(([, , c]) => c())
      return cards
    },
    subtitles: media.flatMap(([, , , s]) => s),
  }
}

const getMediaClipsGetter =
  <F extends FlashcardFields>(
    state: AppState,
    project: ProjectFile,
    media: MediaJson<F>
  ) =>
  () =>
    getMediaClips(state, project, media)
const getMediaCardsGetter =
  <F extends FlashcardFields>(
    state: AppState,
    project: ProjectFile,
    media: MediaJson<F>
  ) =>
  () =>
    getMediaCards(state, project, media)

function getMediaClips<F extends FlashcardFields>(
  state: AppState,
  project: ProjectFile,
  media: MediaJson<F>
): Clip[] {
  return (media.clips || []).map((c): Clip => {
    return {
      id: c.id,
      clipwaveType: 'Primary',
      start: parseFormattedDuration(c.start).asMilliseconds(),
      end: parseFormattedDuration(c.end).asMilliseconds(),
      fileId: media.id,
    }
  })
}
function getMediaCards<F extends FlashcardFields>(
  state: AppState,
  project: ProjectFile,
  media: MediaJson<F>
): Flashcard[] {
  return (media.clips || []).map((c): Flashcard => {
    const flashcardBase = {
      type: project.noteType,
      id: c.id,
      tags: c.tags || [],
      image: c.image && {
        id: c.id,
        type: 'VideoStillImage' as const,
        ...(c.image.time
          ? { seconds: parseFormattedDuration(c.image.time).asSeconds() }
          : null),
      },
    }

    const { fields, cloze } = unescapeClozeFields(c.fields)

    return project.noteType === 'Simple'
      ? {
          ...flashcardBase,
          type: 'Simple',
          fields: { ...blankSimpleFields, ...fields },
          cloze,
        }
      : {
          ...flashcardBase,
          type: 'Transliteration',
          fields: { ...blankTransliterationFields, ...fields },
          cloze,
        }
  })
}

function toMediaSubtitlesRelation(s: SubtitlesJson): MediaSubtitlesRelation {
  switch (s.type) {
    case 'Embedded':
      return {
        type: 'EmbeddedSubtitlesTrack',
        id: s.id,
      }

    case 'External':
      return {
        type: 'ExternalSubtitlesTrack',
        id: s.id,
      }
  }
}
