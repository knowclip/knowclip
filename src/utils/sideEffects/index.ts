import sideEffects from './module'
import { setUpMocks } from '../../preloaded/setUpMocks'

const mocked = setUpMocks('side-effect', sideEffects)

const { uuid, nowUtcTimestamp } = mocked

export { uuid, nowUtcTimestamp }
