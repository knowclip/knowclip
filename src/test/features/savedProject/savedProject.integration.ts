import {
  startApp,
  stopApp,
  TMP_DIRECTORY,
  ASSETS_DIRECTORY,
  GENERATED_ASSETS_DIRECTORY,
  initTestContext,
} from '../../setUpDriver'
import { mockSideEffects } from '../../../utils/sideEffects/mocks'
import { join } from 'path'
import { runAll, step } from '../step'
import { savedProjectTestSteps } from './savedProjectTestSteps'
import { parseProjectJson } from '../../../utils/parseProject'
import { describe, beforeAll, afterAll, test, expect } from 'vitest'

const testId = 'savedProject'

describe('opening and saving a previously saved project', () => {
  const context = initTestContext(testId)

  beforeAll(async () => {
    const { app } = await startApp(context, persistedState)

    await mockSideEffects(app, sideEffectsMocks)
  })

  runAll(
    [
      ...savedProjectTestSteps({
        projectTitle: 'My cool saved project',
        firstClipId: '632a6cff-7fd7-4d0f-b657-0b9636204261',
      }),
      step('resulting project file matches snapshot', async () => {
        test('save project', async () => {
          const actualProjectFileContents = await parseProjectJson(
            join(TMP_DIRECTORY, 'my_previously_saved_project.kyml')
          )

          expect(actualProjectFileContents).toMatchSnapshot()
        })
      }),
    ],
    context
  )

  afterAll(async () => {
    await stopApp(context)
  })
})

const sideEffectsMocks = {
  uuid: [
    '136042f2-00bc-4d9b-a3a7-3da1b7868c78',
    '1e62d23d-3f3b-4785-b8ba-5d61dddc78ed',
    '9a07597c-7885-49bc-97d4-76a2dffdb9aa',
  ],
  nowUtcTimestamp: [
    '2020-01-27T22:28:12Z',
    '2020-01-27T22:28:22Z',
    '2020-02-02T15:31:34Z',
    '2020-02-02T15:31:35Z',
    '2020-02-02T15:31:35Z',
    '2020-02-02T15:31:35Z',
    '2020-02-02T15:31:35Z',
    '2020-02-02T15:31:36Z',
    '2020-02-02T15:31:37Z',
    '2020-02-02T15:31:37Z',
    '2020-02-02T15:31:40Z',
    '2020-02-02T15:31:42Z',
    '2020-02-02T15:31:43Z',
    '2020-02-02T15:31:43Z',
    '2020-02-02T15:31:43Z',
    '2020-02-02T15:31:43Z',
    '2020-02-02T15:31:43Z',
    '2020-02-02T15:31:43Z',
    '2020-02-02T15:31:43Z',
    '2020-02-02T15:31:43Z',
    '2020-02-02T15:31:44Z',
    '2020-03-20T13:27:57Z',
    '2020-03-20T13:27:58Z',
  ],
}
const persistedState: Partial<AppState> = {
  fileAvailabilities: {
    ProjectFile: {
      '91bfd159-155c-4b61-bdd5-d71e2e944773': {
        type: 'ProjectFile',
        id: '91bfd159-155c-4b61-bdd5-d71e2e944773',
        parentId: null,
        name: 'My cool saved project',
        status: 'CURRENTLY_LOADED',
        isLoading: false,
        filePath: join(TMP_DIRECTORY, '/my_previously_saved_project.kyml'),
        lastOpened: '2020-02-01T13:20:55Z',
      },
    },
    MediaFile: {
      '10fd95ee-ba1a-424f-95fb-66a28c91faf1': {
        type: 'MediaFile',
        id: '10fd95ee-ba1a-424f-95fb-66a28c91faf1',
        parentId: '91bfd159-155c-4b61-bdd5-d71e2e944773',
        name: 'polar_bear_cafe.mp4',
        status: 'CURRENTLY_LOADED',
        isLoading: false,
        filePath: join(ASSETS_DIRECTORY, '/polar_bear_cafe.mp4'),
        lastOpened: '2020-02-01T13:21:20Z',
      },
      'bb2bd7fe-4ead-48aa-bad5-88df7a1b5e77': {
        type: 'MediaFile',
        id: 'bb2bd7fe-4ead-48aa-bad5-88df7a1b5e77',
        parentId: '91bfd159-155c-4b61-bdd5-d71e2e944773',
        name: 'piggeldy_cat.mp4',
        status: 'CURRENTLY_LOADED',
        isLoading: false,
        filePath: join(ASSETS_DIRECTORY, '/piggeldy_cat.mp4'),
        lastOpened: '2020-02-01T13:21:08Z',
      },
    },
    ExternalSubtitlesFile: {
      'b3347257-97c4-4c9a-8eae-9bf88a8edc23': {
        type: 'ExternalSubtitlesFile',
        id: 'b3347257-97c4-4c9a-8eae-9bf88a8edc23',
        parentId: '10fd95ee-ba1a-424f-95fb-66a28c91faf1',
        name: 'pbc_jp.ass',
        status: 'CURRENTLY_LOADED',
        isLoading: false,
        filePath: join(ASSETS_DIRECTORY, '/pbc_jp.ass'),
        lastOpened: '2020-02-01T13:21:20Z',
      },
    },
    VttConvertedSubtitlesFile: {
      'd680807d-3a5f-4a25-967f-71871b7d5057': {
        type: 'VttConvertedSubtitlesFile',
        id: 'd680807d-3a5f-4a25-967f-71871b7d5057',
        parentId: '10fd95ee-ba1a-424f-95fb-66a28c91faf1',
        name: 'VttConvertedSubtitlesFile',
        status: 'CURRENTLY_LOADED',
        isLoading: false,
        filePath: join(
          GENERATED_ASSETS_DIRECTORY,
          '/polar_bear_cafe.mp4_2_d680807d-3a5f-4a25-967f-71871b7d5057.vtt'
        ),
        lastOpened: '2020-02-01T13:21:21Z',
      },
      'b3347257-97c4-4c9a-8eae-9bf88a8edc23': {
        type: 'VttConvertedSubtitlesFile',
        id: 'b3347257-97c4-4c9a-8eae-9bf88a8edc23',
        parentId: 'b3347257-97c4-4c9a-8eae-9bf88a8edc23',
        name: 'VttConvertedSubtitlesFile',
        status: 'CURRENTLY_LOADED',
        isLoading: false,
        filePath: join(
          GENERATED_ASSETS_DIRECTORY,
          '/pbc_jp.ass_b3347257-97c4-4c9a-8eae-9bf88a8edc23.vtt'
        ),
        lastOpened: '2020-02-01T13:21:21Z',
      },
    },
    WaveformPng: {
      '10fd95ee-ba1a-424f-95fb-66a28c91faf1': {
        type: 'WaveformPng',
        id: '10fd95ee-ba1a-424f-95fb-66a28c91faf1',
        parentId: '10fd95ee-ba1a-424f-95fb-66a28c91faf1',
        name: 'WaveformPng',
        status: 'CURRENTLY_LOADED',
        isLoading: false,
        filePath: join(
          GENERATED_ASSETS_DIRECTORY,
          '/polar_bear_cafe.mp4_10fd95ee-ba1a-424f-95fb-66a28c91faf1.png'
        ),
        lastOpened: '2020-02-01T13:21:21Z',
      },
      'bb2bd7fe-4ead-48aa-bad5-88df7a1b5e77': {
        type: 'WaveformPng',
        id: 'bb2bd7fe-4ead-48aa-bad5-88df7a1b5e77',
        parentId: 'bb2bd7fe-4ead-48aa-bad5-88df7a1b5e77',
        name: 'WaveformPng',
        status: 'CURRENTLY_LOADED',
        isLoading: false,
        filePath: join(
          GENERATED_ASSETS_DIRECTORY,
          '/piggeldy_cat.mp4_bb2bd7fe-4ead-48aa-bad5-88df7a1b5e77.png'
        ),
        lastOpened: '2020-02-01T13:21:19Z',
      },
    },
    ConstantBitrateMp3: {},
    VideoStillImage: {
      '632a6cff-7fd7-4d0f-b657-0b9636204261': {
        type: 'VideoStillImage',
        id: '632a6cff-7fd7-4d0f-b657-0b9636204261',
        parentId: null,
        name: 'VideoStillImage',
        status: 'CURRENTLY_LOADED',
        isLoading: false,
        filePath: join(
          GENERATED_ASSETS_DIRECTORY,
          '/polar_bear_cafe_10-280_632a6cff-7fd7-4d0f-b657-0b9636204261.png'
        ),
        lastOpened: '2020-02-01T13:20:59Z',
      },
      'a272b282-4d89-4af0-9d17-baa807793f1f': {
        type: 'VideoStillImage',
        id: 'a272b282-4d89-4af0-9d17-baa807793f1f',
        parentId: null,
        name: 'VideoStillImage',
        status: 'CURRENTLY_LOADED',
        isLoading: false,
        filePath: join(
          GENERATED_ASSETS_DIRECTORY,
          '/polar_bear_cafe_16-760_a272b282-4d89-4af0-9d17-baa807793f1f.png'
        ),
        lastOpened: '2020-02-01T13:21:00Z',
      },
      '6fce08e8-9312-4930-9601-c14b470e57ab': {
        type: 'VideoStillImage',
        id: '6fce08e8-9312-4930-9601-c14b470e57ab',
        parentId: null,
        name: 'VideoStillImage',
        status: 'CURRENTLY_LOADED',
        isLoading: false,
        filePath: join(
          GENERATED_ASSETS_DIRECTORY,
          '/polar_bear_cafe_78-180_6fce08e8-9312-4930-9601-c14b470e57ab.png'
        ),
        lastOpened: '2020-02-01T13:21:08Z',
      },
    },
    Dictionary: {},
  },
}
