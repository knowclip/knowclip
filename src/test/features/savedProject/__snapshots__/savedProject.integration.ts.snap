// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`opening and saving a previously saved project > resulting project file matches snapshot > save project 1`] = `
{
  "value": {
    "media": [
      {
        "clips": [
          {
            "end": "00:12.5",
            "fields": {
              "meaning": "Relaxing while eating bamboo grass is the best",
              "pronunciation": "sasa-o tabe-nágara nonbíri-suru-no-wa saikoo-da-naa",
              "transcription": "笹を食べながらのんびりするのは最高だなぁ",
            },
            "id": "632a6cff-7fd7-4d0f-b657-0b9636204261",
            "image": {
              "type": "VideoStill",
            },
            "start": "00:08.04",
            "tags": [
              "pbc",
            ],
          },
          {
            "end": "00:18.4",
            "fields": {
              "meaning": "This kid, lazing about again so early",
              "notes": "\\"Goro-goro\\" is the sound of something big rolling around.",
              "pronunciation": "mata kono ko-wa hirumá-kara góro-goro shite",
              "transcription": "ちょっと  またこの子は昼間からゴロゴロして",
            },
            "id": "a272b282-4d89-4af0-9d17-baa807793f1f",
            "image": {
              "type": "VideoStill",
            },
            "start": "00:15.12",
            "tags": [
              "pbc",
            ],
          },
          {
            "end": "01:21.16",
            "fields": {
              "transcription": "えーと  働かないでお金がもらえる仕事が載っているのはどれですか",
            },
            "id": "6fce08e8-9312-4930-9601-c14b470e57ab",
            "image": {
              "type": "VideoStill",
            },
            "start": "01:15.2",
            "tags": [
              "pbc",
            ],
          },
        ],
        "duration": "01:27.062",
        "flashcardFieldsToSubtitlesTracks": {
          "transcription": "b3347257-97c4-4c9a-8eae-9bf88a8edc23",
        },
        "format": "mov,mp4,m4a,3gp,3g2,mj2",
        "id": "10fd95ee-ba1a-424f-95fb-66a28c91faf1",
        "name": "polar_bear_cafe.mp4",
        "subtitles": [
          {
            "chunksMetadata": {
              "count": 26,
              "endCueMs": 86430,
            },
            "id": "d680807d-3a5f-4a25-967f-71871b7d5057",
            "streamIndex": 2,
            "type": "Embedded",
          },
          {
            "chunksMetadata": {
              "count": 15,
              "endCueMs": 86730,
            },
            "id": "b3347257-97c4-4c9a-8eae-9bf88a8edc23",
            "name": "pbc_jp.ass",
            "type": "External",
          },
        ],
      },
      {
        "clips": [
          {
            "end": "00:08.76",
            "fields": {
              "meaning": "I am not a cat, said Frederick angrily",
              "transcription": "Ich bin keine Katze, sagte Frederick böse",
            },
            "id": "136042f2-00bc-4d9b-a3a7-3da1b7868c78",
            "image": {
              "type": "VideoStill",
            },
            "start": "00:07.02",
            "tags": [
              "piggeldy_cat.mp4",
            ],
          },
          {
            "end": "00:20.0",
            "fields": {
              "meaning": "I didn't say that",
              "transcription": "Das hab' ich nicht gesagt",
            },
            "id": "1e62d23d-3f3b-4785-b8ba-5d61dddc78ed",
            "image": {
              "type": "VideoStill",
            },
            "start": "00:18.42",
            "tags": [
              "piggeldy_cat.mp4",
            ],
          },
        ],
        "duration": "00:58.24",
        "format": "mov,mp4,m4a,3gp,3g2,mj2",
        "id": "bb2bd7fe-4ead-48aa-bad5-88df7a1b5e77",
        "name": "piggeldy_cat.mp4",
      },
    ],
    "project": {
      "createdAt": "2020-02-01T13:20:55Z",
      "id": "91bfd159-155c-4b61-bdd5-d71e2e944773",
      "name": "My cool saved project",
      "noteType": "Transliteration",
      "timestamp": "2020-02-02T15:31:43Z",
    },
  },
}
`;

